src/Block.js

```diff
@@ -0,0 +1,188 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import PropTypes from 'prop-types';
+import AnimateHeight from 'react-animate-height';
+import {Draggable} from 'react-beautiful-dnd';
+import {bindActionCreators} from 'redux';
+import {connect} from 'react-redux';
+import {
+  blockUpdated,
+  deleteBlock,
+  showBlock,
+} from './actions';
+import {
+  getDescendantsIds, getLayout,
+  getNestedBlockDefinition,
+  getSiblingsIds, isSimpleLayout,
+  triggerCustomEvent,
+} from './processing/utils';
+import AddButton from './AddButton';
+import BlockContent from './BlockContent';
+import BlockHeader from './BlockHeader';
+import BlockActions from './BlockActions';
+
+
+@connect((state, props) => {
+  const {fieldId, id} = props;
+  const fieldData = state[fieldId];
+  const blocks = fieldData.blocks;
+  const block = blocks[id];
+  const siblings = getSiblingsIds(state, fieldId, id);
+  const blockDefinition = getNestedBlockDefinition(state, fieldId, id);
+  const hasDescendantError = getDescendantsIds(state, fieldId, id, true)
+    .some(descendantBlockId => blocks[descendantBlockId].hasError);
+  return {
+    blockDefinition,
+    layout: getLayout(blockDefinition, fieldData.isMobile),
+    isSimpleLayout: isSimpleLayout(blockDefinition, fieldData.isMobile),
+    parentId: block.parent,
+    hasError: hasDescendantError,
+    closed: block.closed,
+    hidden: block.hidden,
+    shouldUpdate: block.shouldUpdate,
+    index: siblings.indexOf(id),
+  };
+}, (dispatch, props) => {
+  const {fieldId, id} = props;
+  return bindActionCreators({
+    blockUpdated: () => blockUpdated(fieldId, id),
+    showBlock: () => showBlock(fieldId, id),
+    deleteBlock: () => deleteBlock(fieldId, id),
+  }, dispatch);
+})
+class Block extends React.Component {
+  static propTypes = {
+    fieldId: PropTypes.string.isRequired,
+    id: PropTypes.string.isRequired,
+    standalone: PropTypes.bool,
+    collapsible: PropTypes.bool,
+    sortable: PropTypes.bool,
+    canAdd: PropTypes.bool,
+  };
+
+
+  static defaultProps = {
+    standalone: false,
+    collapsible: true,
+    sortable: true,
+    canAdd: true,
+  };
+
+  constructor(props) {
+    super(props);
+    this.dragHandleRef = React.createRef();
+    this.contentRef = React.createRef();
+  }
+
+  shouldComponentUpdate(nextProps, nextState, nextContext) {
+    return nextProps.shouldUpdate || (nextProps.layout !== this.props.layout);
+  }
+
+  componentDidUpdate(prevProps, prevState, snapshot) {
+    if (!prevProps.shouldUpdate) {
+      this.props.blockUpdated();
+    }
+  }
+
+  triggerCustomEvent(name, data=null) {
+    triggerCustomEvent(ReactDOM.findDOMNode(this), name, data);
```

Comment:
From what I can see below, the `data` parameter is never used.

Reply:
I did this to keep consistency with the other components defining a `triggerCustomEvent` method, like `BlockHeader`.